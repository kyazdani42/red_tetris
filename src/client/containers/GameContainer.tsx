import * as React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router';
import styled from 'styled-components';

import GameController from '../components/Game/Control/GameController';
import Game from '../components/Game/Display/Game';
import LeaveRoomButton from '../components/Game/TopInfo/LeaveRoomButton';
import Spectres from '../components/Game/TopInfo/Spectres';

const BackgroundStyle = styled.div`
  background-color: #000;
  height: 100vh;
`;

const TopInfoContainer = styled.div`
  height: 10vh;
  width: 100%;
  border-bottom: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;

const GameWrapper = styled.div`
  position: relative;
  height: 70vh;
  max-height: 1000px;
  width: 30%;
  min-width: 300px;
  max-width: 500px;
  margin: auto;
  margin-top: 30px;
  border: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.1), 0 5px 5px rgba(255, 55, 55, 0.2);
  background-color: #000;
`;

const ControlContainer = styled.div`
  height: 7vh;
  margin: auto;
  margin-top: 30px;
  width: 30%;
  min-width: 300px;
  max-width: 500px;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: #000;
  border: 1px solid #fff;
`;

interface Props {
  socket: SocketIOClient.Socket | null;
}

const GameContainer: React.SFC<Props> = ({ socket }) => {
  if (!socket) {
    return <Redirect to="/" />;
  }
  return <BackgroundStyle>
      <TopInfoContainer>
        <Spectres />
        <LeaveRoomButton />
      </TopInfoContainer>
      <GameWrapper>
        <Game />
      </GameWrapper>
      <ControlContainer>
        <GameController />
      </ControlContainer>
    </BackgroundStyle>;
};

const mapStateToProps = (state: any) => ({
  socket: state.app.socket
});

export default connect(mapStateToProps)(GameContainer);
