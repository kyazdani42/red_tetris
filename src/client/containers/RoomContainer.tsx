import * as React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import styled from 'styled-components';

import { State } from '../store';

import { Background } from '../components/Background/Background';
import CreateRoomButton from '../components/Home/CreateRoom';
import { RoomRow } from '../components/Home/RoomRow';

interface RoomsProps {
  playerName: string | null;
  rooms: RoomType[];
  socket: SocketIOClient.Socket | null;
}

const RoomWrapper = styled.div`
  position: relative;
`;

const StyledContainer = styled.div`
  font-family: ${props => props.theme.fonts.text};
  font-size: 18px;
  font-weight: 300;
  background-color: #000;
  color: #000;

  z-index: 100;
  height: 80vh;
  width: 70vw;
  margin: 10vh auto auto;
  max-width: 500px;
  border: 1px solid rgba(255,69,100,0.4);
  box-shadow: 0 10px 10px rgba(255,44,55,0.20), 0 5px 5px rgba(255,55,55,0.20);
  transition: all 1s;
  align-self: flex-end;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
`;

const RowWrapperHideOverflow = styled.div`
  height: 88%;
  width: 90%;
  overflow: hidden;
  border: 1px solid rgba(255, 255, 255, 0.1);
`;

const RowWrapper = styled.div`
  height: 100%;
  width: 100%;
  padding-right: 17px;
  overflow-y: scroll;
`;

export const RoomContainer: React.SFC<RoomsProps> = ({ rooms, socket, playerName }) => {
  if (socket) {
    const redirectUrl = `${socket.nsp}[${playerName}]`;
    return <Redirect to={redirectUrl} />;
  } else {
    return <RoomComponent rooms={rooms} />;
  }
};

export const RoomComponent: React.SFC<{ rooms: RoomsProps['rooms'] }> = ({ rooms }) => (
  <React.Fragment>
    <Background />
    <RoomWrapper>
      <StyledContainer>
        <RowWrapperHideOverflow>
          <RowWrapper>
            {getRoomRows(rooms)}
          </RowWrapper>
        </RowWrapperHideOverflow>
        <CreateRoomButton />
      </StyledContainer>
    </RoomWrapper>
  </React.Fragment>
);

export const getRoomRows = (rooms: RoomType[]): JSX.Element[] =>
  rooms.map((d, i) => <RoomRow {...d} key={`room-_${d.id}_${i}`} />);

const mapStateToProps = ({ app: { rooms, socket, playerName } }: State) => ({ rooms, socket, playerName });

export default connect(mapStateToProps)(RoomContainer);
