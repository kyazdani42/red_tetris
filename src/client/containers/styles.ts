import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body, html, #root {
    padding: 0;
    margin: 0;
  }
  @font-face {
    font-family: 'Circular Book';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book');
  }
  @font-face {
    font-family: 'Circular Book Italic';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book Italic');
  }
  @font-face {
    font-family: 'Circular Bold';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Bold');
  }
`;

export const BackgroundStyle = styled.div`
  background-color: #000;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
`;

export const TopInfoContainer = styled.div`
  height: 10%;
  width: 100%;
  border-bottom: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
`;

export const GameWrapper = styled.div`
  position: relative;
  z-index: 11;
  width: 35vmin;
  height: 70vmin;
  min-width: 200px;
  min-height: 400px;
  border: 1px solid #fff;
  box-shadow: 0 10px 10px rgba(255, 44, 55, 0.1), 0 5px 5px rgba(255, 55, 55, 0.2);
  background-color: #000;
`;

export const ControlContainer = styled.div`
  height: 7%;
  width: 30%;
  margin-bottom: 20px;
  min-width: 200px;
  max-width: 400px;
  box-shadow: 0 10px 10px rgba(255,44,55,0.10), 0 5px 5px rgba(255,55,55,0.20);
  background-color: #000;
  border: 1px solid #fff;
`;

export const RoomWrapper = styled.div`
  position: relative;
`;

export const StyledContainer = styled.div`
  font-family: ${props => props.theme.fonts.text};
  font-size: 18px;
  font-weight: 300;
  background-color: #000;
  color: #000;
  z-index: 100;
  height: 80vh;
  width: 70vw;
  margin: 10vh auto auto;
  max-width: 500px;
  border: 1px solid rgba(255,69,100,0.4);
  box-shadow: 0 10px 10px rgba(255,44,55,0.20), 0 5px 5px rgba(255,55,55,0.20);
  transition: all 1s;
  align-self: flex-end;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
`;

export const RowWrapperHideOverflow = styled.div`
  height: 88%;
  width: 90%;
  overflow: hidden;
  border: 1px solid rgba(255, 255, 255, 0.1);
`;

export const RowWrapper = styled.div`
  height: 100%;
  width: 100%;
  padding-right: 17px;
  overflow-y: scroll;
`;
