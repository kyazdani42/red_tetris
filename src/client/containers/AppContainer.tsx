import * as React from 'react';
import { Route } from 'react-router-dom';
import { createGlobalStyle } from 'styled-components';

import { Background } from '../components/Background';
import GameContainer from './GameContainer';
import RoomContainer from './RoomContainer';

const GlobalStyle = createGlobalStyle`
  body, html, #root {
    width: 100%;
    height: 100%;
    padding: 0;
    margin: 0;
  }
  @font-face {
    font-family: 'Circular Book';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book');
  }
  @font-face {
    font-family: 'Circular Book Italic';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Book Italic');
  }
  @font-face {
    font-family: 'Circular Bold';
    font-style: normal;
    font-weight: normal;
    src: local('assets/Circular Bold');
  }
`;

export const App = () => (
  <React.Fragment>
    <Background />
    <Route exact path="/" component={RoomContainer} />
    <Route path="/:room[:player_name]" component={GameContainer} />
    <GlobalStyle />
  </React.Fragment>
);
