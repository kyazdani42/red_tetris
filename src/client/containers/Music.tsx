import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../redux/store';

import { setMusicPlaying } from '../redux/actions';
import { ImgStyle, MusicStyle } from './styles';

interface Props {
  isPlaying: boolean;
  dispatchSetIsPlaying: () => void;
}

export const Music: React.SFC<Props> = ({ isPlaying, dispatchSetIsPlaying }) => {
  return (
    <React.Fragment>
      <MusicStyle
        onClick={() => {
          const audio: any = document.getElementById('tetris-music');
          if (audio && isPlaying) {
            audio.pause();
            dispatchSetIsPlaying();
          } else if (audio) {
            audio.play();
            dispatchSetIsPlaying();
          }
        }}
      >
        <ImgStyle src={isPlaying ? '/assets/iconpause.png' : '/assets/iconplay.png'} />
      </MusicStyle>
      <audio id="tetris-music" loop autoPlay>
        <source src="/assets/theme-tetris.mp3">
        </source>
      </audio>
    </React.Fragment>
  );
};

const mapStateToProps = (state: State) => ({
  isPlaying: state.app.musicPlaying
});

const mapDispatchToProps = (dispatch: any) => ({
  dispatchSetIsPlaying: () => dispatch(setMusicPlaying())
});

export default connect(mapStateToProps, mapDispatchToProps)(Music);
