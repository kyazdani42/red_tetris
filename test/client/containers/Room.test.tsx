// tslint:disable
import { shallow } from 'enzyme';
import * as React from 'react';
import { Redirect } from 'react-router';

import { RoomContainer, RoomComponent, getRoomRows, mapStateToProps } from '../../../src/client/containers/RoomContainer';

describe('testing the component', () => {
  const getProps = (rooms: any, socket: any, playerName: any) => ({
    rooms,
    socket,
    playerName
  });
  it('returns the room component when socket is null', () => {
    const props = getProps([], null, null);
    const wrapper = shallow(<RoomContainer {...props} />);
    expect(wrapper.containsMatchingElement(<RoomComponent rooms={props.rooms} />)).toBeTruthy();
  });
  it('returns a redirection with the expected url when socket is not null', () => {
    const props = getProps([], { nsp: 'test' }, 'player');
    const wrapper = shallow(<RoomContainer {...props} />);
    expect(wrapper.containsMatchingElement(<Redirect to="test[player]" />)).toBeTruthy();
  })
  it('smoke tests the component', () => {
    const props = getProps([{ id: 1 }, { id: 2 }], null, null);
    const wrapper = shallow(<RoomComponent {...props} />);
    expect(wrapper.update()).toMatchSnapshot();
  })
});

describe('testing getRoomRows', () => {
  const getElements = (i: number) => {
    const els = [];
    for (let x = 0; x < i; x++) {
      els.push({ name: 'room' + x, players: 1, ownerName: x.toString(20) as any, running: Boolean(x % 2) });
    }
    return els;
  };
  it('returns the number of elements in the array mapped to nodes', () => {
    const elements = getElements(10);
    const nodes = getRoomRows(elements);
    expect(nodes.length).toBe(10);
  });
  it('checks the first element has the correct attributed key', () => {
    const elements = getElements(1);
    const nodes = getRoomRows(elements);
    expect(nodes[0].key).toEqual('room-_room0_0');
  });
})

describe('testing mapStateToProps', () => {
  const state: any = {
    app: {
      rooms: [],
      socket: 'testSocket',
      playerName: 'testPlayer'
    }
  };
  it('tests the function', () => {
    expect(mapStateToProps(state)).toEqual(state.app);
  })
})
