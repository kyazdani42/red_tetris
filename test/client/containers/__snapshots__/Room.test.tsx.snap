// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    playerName={null}
    rooms={
      Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ]
    }
    socket={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "rooms": Array [
        Object {
          "id": 1,
        },
        Object {
          "id": 2,
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "rooms": Array [
          Object {
            "id": 1,
          },
          Object {
            "id": 2,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
